src/database/sqlite/mod.rs

```diff
@@ -682,6 +696,71 @@ impl SqliteConn {
         .expect("Db must not fail")
     }
 
+    /// Store transactions in database, ignoring any that already exist.
+    pub fn new_txs(&mut self, txs: &[bitcoin::Transaction]) {
+        db_exec(&mut self.conn, |db_tx| {
+            for tx in txs {
+                let txid = &tx.txid()[..].to_vec();
+                let tx_ser = bitcoin::consensus::serialize(tx);
+                db_tx.execute(
+                    "INSERT INTO transactions (txid, tx) VALUES (?1, ?2) \
+                        ON CONFLICT DO NOTHING",
+                    rusqlite::params![txid, tx_ser,],
+                )?;
+            }
+            Ok(())
+        })
+        .expect("Database must be available")
+    }
+
+    pub fn list_wallet_transactions(
+        &mut self,
+        txids: &[bitcoin::Txid],
+    ) -> Vec<DbWalletTransaction> {
+        // The UNION will remove duplicates.
```

Comment:
I'm writing a unit test on the migration from v4 to v5 and it seems like it doesn't.

Reply:
Thanks, I've cherry-picked your commit and made some small changes.